tst/hooks/sdk/useMediaStreamMetrics.test.tsx

```diff
@@ -0,0 +1,191 @@
+
+import { AudioVideoContext } from '../../../src';
+import useMediaStreamMetrics from '../../../src/hooks/sdk/useMediaStreamMetrics';
+
+class NoOPAudioVideoFacade extends DefaultAudioVideoFacade {
+  constructor() {
+    super(
+      new NoOpAudioVideoController(),
+      // @ts-ignore
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined
+    );
+  }
+
+  addObserver(observer: AudioVideoObserver): void {
+    super.addObserver(observer);
+  }
+
+  removeObserver(observer: AudioVideoObserver): void {
+    super.removeObserver(observer);
+  }
+}
+
+function createMeetingSessionConfiguration(): MeetingSessionConfiguration {
+  const configuration = new MeetingSessionConfiguration();
+  configuration.meetingId = 'foo-meeting';
+  configuration.urls = new MeetingSessionURLs();
+  configuration.urls.audioHostURL = 'https://audiohost.test.example.com';
+  configuration.urls.turnControlURL = 'https://turncontrol.test.example.com';
+  configuration.urls.signalingURL = 'https://signaling.test.example.com';
+  configuration.credentials = new MeetingSessionCredentials();
+  configuration.credentials.attendeeId = 'foo-attendee';
+  configuration.credentials.joinToken = 'foo-join-token';
+  configuration.attendeePresenceTimeoutMs = 5000;
+  return configuration;
+}
+
+function createClientMetricReport(): DefaultClientMetricReport {
+  class FakeClientMetricReport extends DefaultClientMetricReport {
+    constructor() {
+      super(new NoOpDebugLogger());
+    }
+
+    getObservableVideoMetrics(): { [p: string]: { [p: string]: {} } } {
+      return {
+        1: {
+          1: {
+            videoDownstreamBitrate: 8000,
+            videoDownstreamPacketLossPercent: 50,
+            videoDownstreamPacketsReceived: 1000,
+          },
+        },
+      };
+    }
+
+    getObservableMetrics(): { [p: string]: number } {
+      return {
+        availableOutgoingBitrate: 1000,
+        availableIncomingBitrate: 1000,
+      };
+    }
+  }
+
+  return new FakeClientMetricReport();
+}
+
+const initialMediaStreamMetrics = {
+  audioPacketsSentFractionLossPercent: null,
+  audioPacketsReceivedFractionLossPercent: null,
+  availableOutgoingBandwidth: null,
+  availableIncomingBandwidth: null,
+  videoStreamMetrics: {},
+};
+
+const expectedMediaStreamMetrics = {
+  audioPacketsSentFractionLossPercent: NaN,
+  audioPacketsReceivedFractionLossPercent: NaN,
+  availableOutgoingBandwidth: 1,
+  availableIncomingBandwidth: 1,
+  videoStreamMetrics: {
+    '1': {
+      '1': {
+        videoDownstreamBitrate: 8000,
+        videoDownstreamPacketLossPercent: 50,
+        videoDownstreamPacketsReceived: 1000,
+      },
+    },
+  },
+};
+
+function createAudioVideoFacade(): DefaultAudioVideoFacade {
+  class TestBackoff implements Backoff {
+    reset(): void {}
+
+    nextBackoffAmountMs(): number {
+      return 1;
+    }
+  }
+
+  const logger = new NoOpDebugLogger();
+  const deviceController = new DefaultAudioVideoController(
```

Comment:
why is deviceController `DefaultAudioVideoController`?

Reply:
```js
act(() => {
  for (const observer of Array.from(
    // @ts-ignore
    audioVideoFacade.audioVideoController.observerQueue <=================
  )) {
    // @ts-ignore
    observer.metricsDidReceive(clientMetricReport);
  }
});
```

To test if the hook can get the metrics via `metricsDidReceive` event.

Reply:
Updated